net/hass/files/hass.init

```diff
@@ -0,0 +1,34 @@
+#!/bin/sh /etc/rc.common
+
+START=50
+USE_PROCD=1
+PROG=/usr/bin/hassd.sh
+PIDFILE=/var/run/hass.pid
+
+start_service() {
+    procd_open_instance hass
+    procd_set_param command $PROG
+    procd_set_param pidfile $PIDFILE
+    procd_close_instance
+}
+
+all_child_pids() {
+    pid=$1
+    echo $pid
+
+    for child_pid in $(pgrep -P $pid); do
+        echo "$(all_child_pids $child_pid)"
+    done
+}
+
+stop_service() {
+    for pid in $(all_child_pids $(cat $PIDFILE)); do
+        kill $pid
+    done
```

Comment:
what did you do that procd isn't handling this for you? consider fixing your scripts that you don't need this sort of procedure to die politely.

Reply:
The way I'm spawning processes, procd does not correctly kill all child processes, that's why I wrote this.